inginious/agent/docker_agent/__init__.py

```diff
@@ -517,6 +569,37 @@ async def _write_to_container_stdin(self, write_stream, message):
         write_stream.write(msg)
         await write_stream.drain()
 
+    async def _handle_student_container_outputs(self, student_read_stream, grading_write_stream):
+        """ Receive outputs (stdout and stderr) from student_container and send them to grading_container without decoding """
+        buffer = bytearray()
```

Comment:
This code is structurally really similar to https://github.com/UCL-INGI/INGInious/pull/700/files#diff-16e5de0847426dadade80d602aeb5d530ad44d13a8b92a94bb720a14abd45392R522

There is probably some refactor possible here to make both cleaner.

Reply:
Indeed, some code duplication was removed there :)